src/main/java/io/ebeaninternal/server/cache/DefaultServerCache.java

```diff
@@ -276,19 +274,24 @@ public void runEviction() {
     long trimmedByTTL = 0;
     long trimmedByLRU = 0;
 
-    ArrayList<CacheEntry> activeList = new ArrayList<>();
+    ArrayList<CacheEntry> activeList = new ArrayList<>(map.size());
 
-    long idleExpire = System.currentTimeMillis() - (maxIdleSecs * 1000);
-    long ttlExpire = System.currentTimeMillis() - (maxSecsToLive * 1000);
+    // Changing to nanotime as mentioned by Roland Praml on the mailinglist/google group
+    
+    // Using startNanos because: it seems that in this situation, the important thing is when the computation started.
+    // Looking at this code I feel like that should be the point of reference.
+    // Calling nanoTime 2 times, even so close, is going to give different results.
+    long idleExpireNano =  startNanos- TimeUnit.SECONDS.toNanos(maxIdleSecs);
+    long ttlExpireNano = startNanos - TimeUnit.SECONDS.toNanos(maxSecsToLive);
```

Comment:
This is an actual change to the functionality of the product. @rPraml and @rbygrave plz have a look when you have the time.